audiodata.cpp

```diff
@@ -179,16 +179,22 @@ namespace KeyFinder {
     }
     std::deque<double>::const_iterator readAt = samples.begin();
     std::deque<double>::iterator writeAt = samples.begin();
+
+    // Prevent std::advance out of iterator range problems
+    size_t numSamplesRemaining = samples.size();
+
     while (readAt < samples.end()) {
       double mean = 0.0;
-      if (shortcut) {
+      if (shortcut && numSamplesRemaining > factor) {
```

Comment:
I think >= would be just fine.  I was iffy on it though, so I just opted for > in my code since I knew it'd always work and only add at most factor # of additional iterator advances/adds.  

By all means go with >= if it looks good to you though.  I'm still pretty unfamiliar with the rest of the code so I just opted for safety